Teleportation/Tests.qs

```diff
@@ -189,6 +189,28 @@ namespace Quantum.Kata.Teleportation {
         TeleportPreparedStateTestLoop(PrepareAndSendMessage_Reference, ReconstructAndMeasureMessage);
     }
     
+    @Test("QuantumSimulator")
+    operation T18_EntanglementSwapping () : Unit {
+        
+        let numRepetitions = 100;
+
+        for i in 0 .. numRepetitions {
+            use (qAlice1, qAlice2) = (Qubit(), Qubit());
+            H(qAlice1);
+            CNOT(qAlice1, qAlice2);
+
+            use (qBob1, qBob2) = (Qubit(), Qubit());
+            H(qBob1);
+            CNOT(qBob1, qBob2);
+            
+            let (teleportOp, adjustOp) = EntanglementSwapping();
+            let result = teleportOp(qAlice1, qBob1);
+            adjustOp(qBob2, result);
+
+            let (c1, c2) = (M(qAlice2) == One, M(qBob2) == One);
```

Comment:
This test is a bit too relaxed to verify that the resulting state of qubits qAlice2 and qBob2 is (|00⟩ + |11⟩) / sqrt(2); it only checks that measurement results match, which would also be the case for the state (|00⟩ - |11⟩) / sqrt(2), or any other superposition of the basis states |00⟩ and |11⟩.

A better way to verify that the state of the qubits is the state you'd expect is the approach used in the Superposition kata (detailed in [this blog post](https://devblogs.microsoft.com/qsharp/inside-the-quantum-katas-part-1/)): apply adjoint of the operation that would prepare these qubits in the required state, and use `AssertAllZero` to check that the state of the qubits afterwards is |0...0⟩.

This will also allow to reduce the number of loop iterations: since now we only need to cover the 4 possible outcomes of the measurement in teleportation process, rather than to gather statistics on the measurement outcomes for a superposition state, we can get away with something like 10-15 iterations instead of a hundred.

Reply:
I should have done this from the beginning - this is obviously a much better way - done